src/server/load_statistics.rs

```diff
@@ -0,0 +1,103 @@
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Arc;
+use std::time::Instant;
+
+use libc::{getpid, pid_t};
+
+use super::server::GRPC_THREAD_PREFIX;
+use util::metrics::{get_thread_ids, Stat};
+
+pub struct ThreadLoad {
+    term: AtomicUsize,
+    load: AtomicUsize,
+    threshold: usize,
+}
+
+impl ThreadLoad {
+    pub fn with_threshold(threshold: usize) -> Self {
+        ThreadLoad {
+            term: AtomicUsize::new(0),
+            load: AtomicUsize::new(0),
+            threshold,
+        }
+    }
+
+    #[allow(dead_code)]
+    pub fn in_heavy_load(&self) -> bool {
+        self.load.load(Ordering::SeqCst) > self.threshold
+    }
+
+    /// Incease when every time updating `load`.
+    #[allow(dead_code)]
+    pub fn term(&self) -> usize {
+        self.term.load(Ordering::SeqCst)
```

Comment:
em, I think you don't need to use SeqCst, Relaxed or Acquire/Releas maybe fine here.